### DISCUSSION TITLE
Como criar um controller em um Post N-N

### DISCUSSION THREAD

**Discussion by mathfrance** [balta-io/forum#241] [Tira Dúvidas] [answered]
Boa noite a minha dúvida é para a seguinte situação:

Classe Game com prop de lista de Platforms
![EntertenimentManagement (Running) - Microsoft Visual Studio](https://user-images.githubusercontent.com/23667292/215004001-b22b8a56-6a41-4b2c-a645-f1912003e001.jpg)

Classe Platform com prop de lista de Games
![EntertenimentManagement (Running) - Microsoft Visual Studio_2](https://user-images.githubusercontent.com/23667292/215004209-230f124a-9f8c-4799-8723-58789fae9ed7.jpg)

Game e Platform possuem relação N-N e o mapeamento no banco originou uma terceira tabela GamePlatforms.

As Platforms já estão inseridas no banco e esse é o post que estou fazendo via Postman:
![PostGame - My Workspace](https://user-images.githubusercontent.com/23667292/215004512-5c46f632-13fd-4130-b680-70322c9ba7b7.jpg)

E assim está o meu controller:
![New Discussion · balta-ioforum - Google Chrome](https://user-images.githubusercontent.com/23667292/215004855-64a72744-e155-4efa-acfa-e8282c818746.jpg)

Criei a lógica da linha 62 a 69 para obter as platforms relacionadas e fazer o Insert na tabela já mapeando o relacionamento na tabela GamePlatforms, porém gostaria de saber se não há uma forma mais eficiente, talvez algum recurso onde ele já monte meu model considerando os Ids de Platforms que passei no Json?

P.s: Se eu não fizer esse tratamento ele tenta inserir no Platform e eu só gostaria que ele inserisse os Ids no GamePlatforms.

DÚVIDA 2:

Em adicional a isso estou tendo um exceção de loop infinito na linha 77, que o prop de uma classe fica chamando o prop da outra infinitamente, estourando a seguinte exceção:

"System.Text.Json.JsonException: A possible object cycle was detected. This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 32. Consider using ReferenceHandler.Preserve on JsonSerializerOptions to support cycles. Path: $.Data.Platforms.Games.Platforms.Games.Platforms.Games.Platforms.Games.Platforms.Games.Platforms.Games.Platforms.Games.Platforms.Description."

Não entendi o pq. Caso possa me esclarecer por gentiliza serei grato.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @mathfrance, como vai?

"Game e Platform possuem relação N-N e o mapeamento no banco originou uma terceira tabela GamePlatforms."
- É assim mesmo, o banco sempre cria uma tabela associativa e traz a primary key dos dois relacionamentos.

Na sua questão, pelo que entendi, você tem o Id das plataformas mas é obrigado a preencher elas indo no banco um a um pois não tem uma propriedade `PlatformId` (Propriedade de referência) no seu modelo. Se você, além das referências, manter o Id, fica mais fácil. Acredito que comento sobre isto no curso de EF.

Outra opção, seria receber uma lista de Ids (Plataformas) e preencher o Game com estes Ids, mas isso via query mesmo, seria a forma mais otimizada.

E uma tarceira forma, seria fazer um select e já buscar todas as plataformas, e depois só passar elas para o `model.Platform`.

Sobre sua segunda dúvida, não é loop infinito e sim um problema de referência cíclica com a serialização do JSON. Eu mostro como resolver isto no curso da ASP.NET.
